### PR TITLE
Tee stdout to a log file next to the robot code

### PR DESCRIPTION
**Pull request by PeterJCLaw**
This is useful when debugging issues with the simulator as well as hopefully generally useful for the teams.

### CODE REVIEW

File: controllers/sr_controller/sr_controller.py [hunk 1]
```diff
@@ -137,27 +140,65 @@ def reconfigure_environment(robot_file: Path) -> None:
     os.chdir(str(robot_file.parent))
 
 
+def log_filename(zone_id: int) -> str:
+    now = datetime.datetime.now()
```
Comment (prophile) [inline controllers/sr_controller/sr_controller.py:144]: Can we have a comment here notating that this is explicitly and deliberately both naïve and in the local timezone?

File: controllers/sr_controller/sr_controller.py [hunk 2]
```diff
@@ -137,27 +140,65 @@ def reconfigure_environment(robot_file: Path) -> None:
...
+
+
+class TeeStdout:
+    def __init__(self, name: Path) -> None:
+        self.name = name
+
+    def __enter__(self) -> None:
+        self.file = open(str(self.name), mode='w').__enter__()
+        self.stdout = sys.stdout
+        sys.stdout = self  # type: ignore
```
Comment (prophile) [inline controllers/sr_controller/sr_controller.py:158]: We can use the `redirect_stdout` context manager for this.

File: controllers/sr_controller/sr_controller.py [hunk 3]
```diff
@@ -137,27 +140,65 @@ def reconfigure_environment(robot_file: Path) -> None:
...
-    os.environ['SR_ROBOT_MODE'] = robot_mode
-    os.environ['SR_ROBOT_FILE'] = str(robot_file)
+        # Pass through the various data our library needs
+        os.environ['SR_ROBOT_ZONE'] = str(robot_zone)
+        os.environ['SR_ROBOT_MODE'] = robot_mode
+        os.environ['SR_ROBOT_FILE'] = str(robot_file)
 
-    # Swith to running the competitor code
-    reconfigure_environment(robot_file)
+        # Swith to running the competitor code
```
Comment (prophile) [inline controllers/sr_controller/sr_controller.py:198]: Typo: Swith

Comment (PeterJCLaw) [reply controllers/sr_controller/sr_controller.py:204]: :( I'll fix that separately though.

File: controllers/sr_controller/sr_controller.py (hunk 1 above)
Comment (PeterJCLaw) [reply controllers/sr_controller/sr_controller.py:146]: log_filename

File: controllers/sr_controller/sr_controller.py (hunk 2 above)
Comment (PeterJCLaw) [reply controllers/sr_controller/sr_controller.py]: 5d62b19

File: controllers/sr_controller/sr_controller.py (hunk 1 above)
Comment (PeterJCLaw) [reply controllers/sr_controller/sr_controller.py:146]: 3f376f1

**Pull request merged**